TodoListManagerTest/src/test/java/PerformanceTest/TodosPerformanceTest.java

```diff
@@ -0,0 +1,177 @@
+        int targetIndex = 0;
+
+        for(int i=1; i<=max;i++){
+            JSONObject body = new JSONObject();
+            String newTitle = "New Todo " + i;
+            String newDescription = "New Description";
+            boolean newDoneStatus = false;
+            body.put("title",newTitle);
+            body.put("doneStatus",newDoneStatus);
+            body.put("description",newDescription);
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).post(pUrl);
+        }
+
+        for(int i=1; i<=max;i++){
+            JSONObject body = new JSONObject();
+            String newTitle = "New Todo " + i;
+            String newDescription = "New Description";
+            boolean newDoneStatus = true;
+            body.put("title",newTitle);
+            body.put("doneStatus",newDoneStatus);
+            body.put("description",newDescription);
+            long startTime = System.nanoTime();
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).put(pUrl+"/"+i);
+            long endTime = System.nanoTime();
+            if(i == populationSize[targetIndex]){
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize();
+                time[targetIndex] = (endTime-startTime);
+                cpuUsage[targetIndex] = cpuLoad;
+                freeMemory[targetIndex] = memory;
+                targetIndex++;
+            }
+        }
+        System.out.println("---Update TODOS---");
+        for (int k = 0; k < populationSize.length; k++) {
+            System.out.println("Time for size " + populationSize[k] + ": " + time[k] + "ns");
+            System.out.println("CPU usage for size " + populationSize[k] + ": " + cpuUsage[k] + "%");
+            System.out.println("Available free memory for size " + populationSize[k] + ": " + freeMemory[k] + "bytes");
+        }
+    }
+
+    @Test
+    public void DeleteTodos(){
+        int max = populationSize[populationSize.length - 1];
+        long[] time = new long[populationSize.length];
+        double[] cpuUsage = new double[populationSize.length];
+        long[] freeMemory = new long[populationSize.length];
+        int targetIndex = 0;
+
+        for(int i=1; i<=max;i++){
+            JSONObject body = new JSONObject();
+            String newTitle = "New Todo " + i;
+            String newDescription = "New Description";
+            boolean newDoneStatus = false;
+            body.put("title",newTitle);
+            body.put("doneStatus",newDoneStatus);
+            body.put("description",newDescription);
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).post(pUrl);
+        }
+
+        for(int i=1; i<=max;i++){
+            long startTime = System.nanoTime();
+            Response response = RestAssured.given().delete(pUrl+"/"+i);
+            long endTime = System.nanoTime();
```

Comment:
Here too, in this case you populated 1000 objects and then you are deleting everything afterwards. The correct path should be you populate ... and once population reach one target then you delete one todo and check metrics, and then so on

Reply:
another way to do it is the same as what jialin did, so you have to reverse the for loop since right now the time that you are getting for 1 is actually the time for target 1000

Reply:
Done